misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/util/SyntaxTreeUtils.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.semver.checker.util;
+
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.ServiceDeclarationNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.Token;
+
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.ballerina.semver.checker.util.PackageUtils.capitalize;
+
+/**
+ * Syntax tree API related utilities.
+ *
+ * @since 2201.2.0
+ */
+public class SyntaxTreeUtils {
+
+    /**
+     * Returns the node typename of the given {@link SyntaxKind} instance.
+     *
+     * @param nodeKind node kind
+     * @return node typename
+     */
+    public static String getNodeKindName(SyntaxKind nodeKind) {
+        return nodeKind.name().toLowerCase(Locale.ROOT).toLowerCase(Locale.getDefault()).replaceAll("_", " ");
```

Comment:
I'm afraid that why we can't use `Locale.ENGLISH` there?

Reply:
Yes `Locale.ENGLISH` would be a safer option for the `toLowerCase` method. Updated the PR.